README.md

```diff
@@ -76,6 +82,95 @@ fastify.register(require('fastify-rate-limit'), {
 })
 ```
 
+
+### Options on the endpoint itself
+
+You can overide the default value set during the registration of the plugin from the route declaration, the values will be used `only` for this route.
+
+Regarding the `whitelist`. There are 2 types :
+ - global : will affect all the endpoint.
+ - endpoint : will affect only the targeted endpoint.
+
+`The Global whitelist` is set in the plugin when registering it with fastify.register(...).
+
+`The Endpoint whitelist` is set on the endpoint directly with the { config : rateLimit : whitelist : [] }object.
+
+you need to set the option `whiteListInRedis` to `true` when registering the plugin. This is only available when using redis. 
+
+
+The ACL is checking based on the value of the key from the keyGenerator. In this example we are checking the Ip but it could be a whitelist of specific user id.
+```js
+const fastify = require('fastify')()
+
+fastify.register(require('fastify-rate-limit'),
+  {
+    global : false,
+    max: 3000, // default global max rate limit
+    whitelist: ['192.168.0.10'], // global whitelist access. 
+    redis: redis, // connection to redis
+    whiteListInRedis: true, // will store the whitelist and check in redis instead of array.
+    skipOnError: false // default false
+    // keyGenerator: function(req) { /* ... */ }, // default (req) => req.raw.ip
+  })
+
+fastify.get('/', {
+  config: {
+    rateLimit: {
+      max: 3,
+      timeWindow: '1 minute'
+    }
+  }
+}, (req, reply) => {
+  reply.send({ hello: 'from ... root' })
+})
+
+fastify.get('/private', {
+  config: {
+    rateLimit: {
+      max: 3,
+      whitelist: ['127.0.2.1', '127.0.3.1'],
+      timeWindow: '1 minute'
+    }
+  }
+}, (req, reply) => {
+  reply.send({ hello: 'from ... private' })
+})
+
+fastify.get('/public', (req, reply) => {
+  reply.send({ hello: 'from ... public' })
+})
+
+fastify.get('/public/sub-rated-1', {
+  config: {
+    rateLimit: {
+      timeWindow: '1 minute',
+      whitelist: ['127.0.0.1'],
+<<<<<<< HEAD
+      onExceeding: function (req) {},
+=======
+      onExceeding: function (req) {
+        console.log('callback on exceededing ... executed before response to client')
+      },
+>>>>>>> 6c3daf23c35af4d1d62ff685699f2e466386424a
```

Comment:
This should be fixed.

Reply:
Yes sorry .. problem during the merge ...